pyomo/core/expr/current.py

```diff
@@ -0,0 +1,128 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and 
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain 
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+from __future__ import division
+import math
+import copy
+
+#
+# Data and methods that are exposed when importing pyomo.core.expr
+#
+__public__ = ['log', 'log10', 'sin', 'cos', 'tan', 'cosh', 'sinh', 'tanh',
+           'asin', 'acos', 'atan', 'exp', 'sqrt', 'asinh', 'acosh',
+           'atanh', 'ceil', 'floor']
+
```

Comment:
Why are we inventing a nonstandard `__public__` module attribute?  This seems undesirable, as "`__`" methods are generally reserved for the interpreter...

Reply:
Agreed